docs/fsharp-cheatsheet.md

```diff
@@ -740,6 +743,112 @@ Another way of implementing interfaces is to use *object expressions*.
 
 *Partial active patterns* share the syntax of parameterized patterns but their active recognizers accept only one argument.
 
+<div id="asynchronous-programming"></div>
+
+## Asynchronous Programming
+
+F# asynchronous programming is centered around two core concepts: async computations and tasks.
+
+    async {
+        do! Async.Sleep (waitInSeconds * 1000)
+        let! asyncResult = asyncComputation
+        use! disposableResult = iDisposableAsyncComputation
+    }
+
+### Async vs Tasks
+
+An async computation is a unit of work, and a task is a promise of a result. A subtle but important distinction.
+Async computations are composable and are not started until explicitly requested; Tasks (when created using the `task` expression) are _hot_:
+
+    let runAsync waitInSeconds =
+        async {
+            printfn "Created Async"
+            do! Async.Sleep (waitInSeconds * 1000)
+            printfn $"Completed Async"
+        }
+        
+    let runTask waitInSeconds =
+        task {
+            printfn "Started Task"
+            do! System.Threading.Tasks.Task.Delay (waitInSeconds * 1000)
+            printfn $"Completed Task"
+        }
+    
+    let asyncComputation = runAsync 5 // returns Async<unit> and does not print anything
+    let newTask = runTask 3  // returns System.Threading.Tasks.Task<unit> and outputs: "Started Task" <3 second delay> "Completed Task"
+
+    asyncComputation |> Async.RunSynchronously  // this now runs the async computation
+    newTask.Wait() // this will have already completed by this point
```

Comment:
wait is always bad news, so not a good thing to show in a cheat sheet that people are going to blindly grab things from

Reply:
What would you recommend here?